packages/contracts-ts/scripts/generate-addresses.ts

```diff
@@ -0,0 +1,254 @@
+import glob from 'glob'
+import fs from 'fs'
+import path from 'path'
+import { z } from 'zod'
+import { Abi as AbiValidator, Address as AddressValidator } from 'abitype/zod'
+import { fileURLToPath } from 'url'
+
+const __dirname = path.dirname(fileURLToPath(import.meta.url))
+
+/**
+ * [zod](https://github.com/colinhacks/zod) validator for how the deployments json files
+ * are expected to be shaped
+ */
+const deploymentValidator = z.object({
+  address: AddressValidator,
+  abi: AbiValidator
+})
+type Deployment = z.infer<typeof deploymentValidator>
+type Address = z.infer<typeof AddressValidator>
+
+/**
+ * Predeployed contract addresses
+ * In future it would be nice to have a json file in contracts bedrock be generated as source of truth
+ * Keep this in sync with op-bindings/predeploys/addresses.go in meantime
+ */
+const predeployContracts = {
+  LegacyMessagePasser: {
+    address: '0x4200000000000000000000000000000000000000',
+    introduced: 'Legacy',
+    deprecated: true,
+    proxied: true,
+  },
+  DeployerWhitelist: {
+    address: '0x4200000000000000000000000000000000000002',
+    introduced: 'Legacy',
+    deprecated: true,
```

Comment:
Why does this have multiple fields? `deprecated` isn't used, it looks like only the address is needed. The current source of truth of the predeploy addresses is in `core-utils` for typescript